storages/backend/backend.go

```diff
@@ -60,24 +62,44 @@ func (b *Backend) collectMetrics(resp *http.Response, err error, since time.Time
 // Response helps handle responses
 type Response struct {
 	Response *http.Response
+	Request  *http.Request
 	Error    error
 	Backend  *Backend
 }
 
 // DiscardBody drain and close response Body, so connections are properly closed
 func (br *Response) DiscardBody() error {
+	backendName := "unknown"
+	if br.Backend != nil {
+		backendName = br.Backend.Name
+	}
 	if br.Response == nil || br.Response.Body == nil {
+		log.Debugf("ResponseBody for request %s is nil so cannot be closed - backend: %s", br.ReqID(), backendName)
 		return nil
 	}
 	_, err := io.Copy(ioutil.Discard, br.Response.Body)
 	if err != nil {
 		log.Printf("Discard body error %s", err)
 		return err
```

Comment:
Don't we have to close the body here as well? Or at least try to close it? (if the drain fails)